### ISSUE TITLE
parameter settings for SVHN

### ISSUE THREAD

**Issue by sweetTT** [google-research/uda#19]
Hello!If I run SVHN dataset on GPU, which value should I set for train_steps, I use train_steps of 200000, It seems that the model is overfitting.The best accuracy is 96.43 at step 110000.
The following is parameter settings during my training:

task_name=svhn
python main.py \
  --use_tpu=False \
  --do_eval_along_training=True\
  --do_train=True \
  --do_eval=True \
  --task_name=${task_name} \
  --sup_size=1000 \
  --unsup_ratio=5 \
  --data_dir=data/proc_data/${task_name} \
  --model_dir=ckpt/svhn_gpu \
  --max_save=20   \
  --curr_step=0   \
  --iterations=10000 \
  --train_steps=200000 \
  $@

**Comment by qizhex** [comment]
Hi, I haven't tried SVHN on GPUs before. Will run a few experiments and let you know.

**Comment by sweetTT** [comment]
Thanks!

**Comment by qizhex** [comment]
Hi, I tried a few hyperparameters and got accuracy similar to yours but I can reproduce the performance on TPUs. The main difference between hyperparameters in GPUs and TPUs is that we don't use TSA and use a smaller batch size for unlabeled data on GPUs. We found TSA to be quite effective on TPUs but not so helpful on GPUs. My current hypothesize is that the model needs to see enough unlabeled data after the threshold of TSA is getting very high, i.e., when the model fully uses the labeled data (This is more easily achieved when we have a large batch size for unlabeled data). Otherwise, the model is not fully propagating the information of the labeled data to unlabeled data. Maybe we'll need to develop a variant of TSA to make it work well in the case for the small batch size case.

Feel free to let me know if you can get better results on GPU.

**Comment by sweetTT** [comment]
Hi, changing learning_rate from 0.03 to 0.05, I get accuracy 97.01 at step 170000(total step is 200000).  It is strange that the best accuracy can be got at the end of the training on cifar10, but on svhn the model is always overfitting.

**Comment by qizhex** [comment]
Increasing weight_decay_rate can help with the overfitting issue. For example, 7e-4 works for our experiments on TPUs for SVHN. Not sure how it works on GPU though.

**Comment by varunnair18** [comment]
Hello @qizhex, 

I wanted to follow up with you on your hypothesis for why TSA may not be helpful on GPUs. You stated that the model needs to see enough unlabeled data after the threshold of TSA becomes high, or when the model begins to train on all of the labeled data.

In terms of variants of TSA that could confirm that, wouldn't a schedule similar to the log-schedule that increases more rapidly, or a schedule similar to the log-schedule that plateaus for a longer period help with that issue? (i.e. tuning the scale value for the log schedule, see line 187 of image/main.py )

**Comment by qizhex** [comment]
@varunnair18 Yes, that's a good idea. I am currently traveling and working on my thesis proposal so I might not be able to run many experiments on this.

**Comment by varunnair18** [comment]
No worries @qizhex, if I'm able to improve the results using a modified TSA schedule I will let you know.

**Comment by qizhex** [comment]
Sorry that it has been a while. I updated the code and script to reproduce the SVHN performance on GPUs.
